pandas/io/formats/style_render.py

```diff
@@ -950,15 +956,26 @@ def concatenated_visible_rows(obj):
             visible_index_levels: list[int] = [
                 i for i in range(index_levels) if not self.hide_index_[i]
             ]
+            target_index_levels: list[int] = [
+                i
+                for i in range(index_levels)
+                if "invisible" in clines or not self.hide_index_[i]
```

Comment:
Could you make this an `if` statement within the loop that results in a `continue`? i.e. avoid materializing this list